.github/workflows/codeql-workflow.yml

```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+      - develop
+  pull_request:
+    branches:
+      - master
+      - develop
+    paths-ignore:
+      - '.github/workflows/*.yml'
+
+jobs:
+  job_build_debug_nano_s:
```

Comment:
You should build app for every device. Moreover, job name is misleading. You should keep the default names proposed when setting up the workflow:

```yaml
jobs:
  analyze:
    name: Analyze
```

Reply:
Done